### PR TITLE
AP_Proximity: Removed AP_Proximity_LightWareSF40C_v09

### LINKED ISSUE #19697: AP_Proximity: remove AP_Proximity_LightWareSF40C_v09
We are running out of flash space on some boards so let's remove the older lightware SF40c driver. https://github.com/ArduPilot/ardupilot/blob/master/libraries/AP_Proximity/AP_Proximity_LightWareSF40C_v09.h

FYI @muramura

### PR DESCRIPTION
**Pull request by khanasif786**
This Fixes #19697   Deleted older drivers for fixing running out of flash space in some boards

### CODE REVIEW

**Comment by rmackay9** [comment]
Thanks for this.  I think we should remove the value from the enum and from the RNGFNDx_TYPE parameter description.  Also instead of instantiating the new driver I think we shouldn't try to instantiate a driver at all.

File: libraries/AP_Proximity/AP_Proximity.cpp [hunk 1]
```diff
@@ -282,9 +281,9 @@ void AP_Proximity::detect_instance(uint8_t instance)
...
         return;
     case Type::SF40C_v09:
-        if (AP_Proximity_LightWareSF40C_v09::detect()) {
+        if (AP_Proximity_LightWareSF40C::detect()) {
             state[instance].instance = instance;
-            drivers[instance] = new AP_Proximity_LightWareSF40C_v09(*this, state[instance]);
+            drivers[instance] = new AP_Proximity_LightWareSF40C(*this, state[instance]);
             return;
         }
         break;
```
Comment (hendjoshsr71) [inline libraries/AP_Proximity/AP_Proximity.cpp:289]: If we are removing the driver completely we should remove the `case Type::SF40C_v09:` completely.

For the enumeration in `AP_Proximty.h` we can do something like the following to let others know they cannot use that slot aka Type = 1,
```cpp
    enum class Type {
        None    = 0,
        // 1 was SF40C_v09
```

**Review by hendjoshsr71 — CHANGES_REQUESTED** [review]

File: libraries/AP_Proximity/AP_Proximity.cpp (hunk 1 above)
Comment (khanasif786) [reply libraries/AP_Proximity/AP_Proximity.cpp]: Thanks, @rmackay9 sir for review and @hendjoshsr71 sir for help. I have done suggested changes, please have a look.

**Review by hendjoshsr71 — CHANGES_REQUESTED** [review]
@khanasif786  You still need to remove the following from the `RNGFNDx_TYPE parameter description` (I missed it too at first without Randy's comment ;) Because it is named differently.)

```cpp
,1:LightWareSF40C-legacy
```

This is located here https://github.com/ArduPilot/ardupilot/blob/c21ec13719cd4fbbf94945d756ac5720e2f30b6d/libraries/AP_Proximity/AP_Proximity.cpp#L40

The reason we remove into and do not reorder all the numbers and the matching enumeration, `enum class Type`, is that then we would have to perform parameter conversions. And doing that while not creating a mess for users is quite difficult in this case.


Also when you are done by sure to have single commit. So `squash` all your commits into one, `AP_Proximity: Removed AP_Proximity_LightWareSF40C_v09`

If you need help with `squash` let me know.  https://ardupilot.org/dev/docs/git-interactive-rebase.html

**Comment by khanasif786** [comment]
@hendjoshsr71 Sir, Thanks for you detailed explanation. I have Done suggested changes.

**Review by hendjoshsr71 — APPROVED** [review]
Awesome!

LGTM

@rmackay9  for review

**Comment by rmackay9** [comment]
Merged, thanks!

**Pull request merged**